### PR TITLE
Update to Akka HTTP 10.2.0

### PR DESCRIPTION
**Pull request by jrudolph**
Follow up to #937. Currently only as fodder to the validation.

### CODE REVIEW

**Comment by raboof** [comment]
@jrudolph we can probably undraft this?

**Comment by jrudolph** [comment]
> @jrudolph we can probably undraft this?

From my side, yes. I guess it depends on what the plans are for Akka gRPC regarding keeping support for Akka HTTP 10.1.x?

**Comment by raboof** [comment]
> I guess it depends on what the plans are for Akka gRPC regarding keeping support for Akka HTTP 10.1.x?

I think we should update to Akka HTTP 10.2.x and Akka 2.6.x in Akka gRPC 1.1.0, but we haven't set a timeline for that yet.

**Comment by jrudolph** [comment]
> I think we should update to Akka HTTP 10.2.x and Akka 2.6.x in Akka gRPC 1.1.0, but we haven't set a timeline for that yet.

Sounds good!

So, should this go in an extra branch, then? Or will we branch off for 1.0.x releases if needed?

File: plugin-tester-java/src/test/scala/example/myapp/helloworld/ErrorReportingSpec.scala [hunk 1]
```diff
@@ -32,9 +34,14 @@ class ErrorReportingSpec extends AnyWordSpec with Matchers with ScalaFutures wit
...
     val handler = GreeterServiceHandlerFactory.create(new GreeterServiceImpl(mat), sys)
     val binding = {
-      import akka.http.javadsl.{ ConnectHttp, Http }
+      import akka.http.javadsl.Http
+      import akka.http.javadsl.model.{ HttpRequest, HttpResponse }
 
-      Http(sys).bindAndHandleAsync(handler, ConnectHttp.toHost("127.0.0.1", 0), mat).toCompletableFuture.get
+      Http(sys)
+        .newServerAt("127.0.0.1", 0)
+        .bind((req => handler(req)): akka.japi.function.Function[HttpRequest, CompletionStage[HttpResponse]])
```
Comment (raboof) [inline plugin-tester-java/src/test/scala/example/myapp/helloworld/ErrorReportingSpec.scala:42]: I think we can (and should) make this simpler (#1101), but we can do that in a separate PR

**Review by raboof — APPROVED** [review]
